src/V3Tristate.cpp

```diff
@@ -1877,9 +1870,20 @@ class TristateVisitor final : public TristateBaseVisitor {
 
 public:
     // CONSTRUCTORS
-    explicit TristateVisitor(AstNode* nodep) {
+    explicit TristateVisitor(AstNetlist* netlistp) {
         m_tgraph.clear();
-        iterate(nodep);
+        iterate(netlistp);
+#ifdef VL_LEAK_CHECKS
+        // It's a bit chaotic up there
+        std::vector<AstNode*> unusedRootps;
+        netlistp->foreach([&](AstNode* nodep) {
+            AstNode* const enp = nodep->user1p();
+            if (!enp) return;
+            if (enp->backp()) return;
+            unusedRootps.emplace_back(enp);
+        });
+        for (AstNode* const nodep : unusedRootps) VL_DO_DANGLING(nodep->deleteTree(), nodep);
+#endif
```

Comment:
This whole V3Tristate eagerly creating nodes then not use them makes it all a bit intractable, hence this hack.